cirq/contrib/noise_models/noise_models.py

```diff
@@ -211,54 +212,142 @@ def noisy_moment(self, moment: 'cirq.Moment',
+        error mode from the noise model, leave its map blank when initializing
+        this object.
 
         Args:
             depol_prob_map: Map of depolarizing probabilities for each qubit.
+            bitflip_prob: Probability of a bit-flip during measurement.
+            decay_prob: Probability of T1 decay during measurement.
+                Bitflip noise is applied first, then amplitude decay.
         """
-        for qubit, depol_prob in depol_prob_map.items():
-            value.validate_probability(depol_prob, f'depol prob of {qubit}')
+        for prob_map, desc in [(depol_prob_map, "depolarization prob"),
+                               (bitflip_prob_map, "readout error prob"),
+                               (decay_prob_map, "readout decay prob")]:
+            if prob_map:
+                for qubit, prob in prob_map.items():
+                    value.validate_probability(prob, f'{desc} of {qubit}')
         self.depol_prob_map = depol_prob_map
+        self.bitflip_prob_map = bitflip_prob_map
+        self.decay_prob_map = decay_prob_map
 
     def noisy_moment(self, moment: 'cirq.Moment',
                      system_qubits: Sequence['cirq.Qid']):
-        if (_homogeneous_moment_is_measurements(moment) or
-                self.is_virtual_moment(moment)):
+        if self.is_virtual_moment(moment):
             return moment
+        moments = []
+        if _homogeneous_moment_is_measurements(moment):
+            if self.decay_prob_map:
+                moments.append(
+                    ops.Moment(
+                        ops.AmplitudeDampingChannel(self.decay_prob_map[q])(q)
+                        for q in system_qubits))
+            if self.bitflip_prob_map:
+                moments.append(
+                    ops.Moment(
+                        ops.BitFlipChannel(self.bitflip_prob_map[q])(q)
+                        for q in system_qubits))
+            moments.append(moment)
+            return moments
         else:
-            gated_qubits = [
-                q for q in system_qubits if moment.operates_on_single_qubit(q)
-            ]
-            return [
-                moment,
-                ops.Moment(
-                    ops.DepolarizingChannel(self.depol_prob_map[q])(q)
-                    for q in gated_qubits)
-            ]
-
-
-def simple_noise_from_calibration_metrics(calibration: engine.Calibration
+            moments.append(moment)
+            if self.depol_prob_map:
+                gated_qubits = [
+                    q for q in system_qubits
+                    if moment.operates_on_single_qubit(q)
+                ]
+                if gated_qubits:
+                    moments.append(
+                        ops.Moment(
+                            ops.DepolarizingChannel(self.depol_prob_map[q])(q)
+                            for q in gated_qubits))
+            return moments
+
+
+def simple_noise_from_calibration_metrics(calibration: engine.Calibration,
```

Comment:
can we put this in cirq.google?  Does this cause a problem making a dependency from engine to contrib?  If so we should probably create contrib.google

Reply:
The top-level `__init__.py` prohibits depending on `contrib`:
https://github.com/quantumlib/Cirq/blob/626e16e41d12098e201ce48a4bb25c047cc86bcd/cirq/__init__.py#L53-L55

We _could_ move both this method and the `PerQubit...` type to `cirq.google`, though - that might be cleaner than adding a new directory just for this type. What do you think?

Reply:
Discussed offline - I added a TODO to move this later on.